rule/string_format.go

```diff
@@ -43,10 +47,9 @@ func (StringFormatRule) ParseArgumentsTest(arguments lint.Arguments) *string {
 	// Parse the arguments in a goroutine, defer a recover() call, return the error encountered (or nil if there was no error)
 	go func() {
 		defer func() {
-			err := recover()
+			err := w.parseArguments(arguments)
```

Comment:
Since we changed to return `error` instead of panicking, the `ParseArgumentsTest` became unnecessary. We can remove it.